src/librustc_borrowck/borrowck/mir/dataflow/mod.rs

```diff
@@ -0,0 +1,1168 @@
+                let stmts_len = statements.len();
+                self.flow_state.operator.terminator_effect(&self.ctxt, sets, bb, stmts_len);
+            }
+        }
+    }
+}
+
+fn on_all_children_bits<Each>(path_map: &PathMap,
+                              move_paths: &MovePathData,
+                              move_path_index: MovePathIndex,
+                              mut each_child: Each)
+    where Each: FnMut(MoveOutIndex)
+{
+    return on_all_children_bits_recur(
+        path_map, move_paths, move_path_index, &mut each_child);
+
+    fn on_all_children_bits_recur<Each>(path_map: &PathMap,
+                                        move_paths: &MovePathData,
+                                        move_path_index: MovePathIndex,
+                                        each_child: &mut Each)
+        where Each: FnMut(MoveOutIndex)
+    {
+        // 1. invoke `each_child` callback for all moves that directly
+        //    influence path for `move_path_index`
+        for move_index in &path_map[move_path_index] {
+            each_child(*move_index);
+        }
+
+        // 2. for each child of the path (that is named in this
+        //    function), recur.
+        //
+        // (Unnamed children are irrelevant to dataflow; by
+        // definition they have no associated moves.)
+        let mut next_child_index = move_paths[move_path_index].first_child;
+        while let Some(child_index) = next_child_index {
+            on_all_children_bits_recur(path_map, move_paths, child_index, each_child);
+            next_child_index = move_paths[child_index].next_sibling;
+        }
+    }
+}
+
+impl<'b, 'a: 'b, 'tcx: 'a, BD> PropagationContext<'b, 'a, 'tcx, BD>
+    where BD: BitDenotation, BD::Ctxt: HasMoveData<'tcx>
+{
+    fn reset(&mut self, bits: &mut [usize]) {
+        let e = if BD::initial_value() {usize::MAX} else {0};
+        for b in bits {
+            *b = e;
+        }
+    }
+
+    fn walk_cfg(&mut self, in_out: &mut [usize]) {
+        let mir = self.builder.mir;
+        for (bb_idx, bb_data) in mir.basic_blocks.iter().enumerate() {
+            let builder = &mut self.builder;
+            {
+                let sets = builder.flow_state.sets.for_block(bb_idx);
+                debug_assert!(in_out.len() == sets.on_entry.len());
+                in_out.clone_from_slice(sets.on_entry);
+                bitwise(in_out, sets.gen_set, &Union);
+                bitwise(in_out, sets.kill_set, &Subtract);
+            }
+            builder.propagate_bits_into_graph_successors_of(in_out,
+                                                            &mut self.changed,
+                                                            (repr::BasicBlock::new(bb_idx), bb_data));
+        }
+    }
+}
+
+impl<'a, 'tcx: 'a, BD> MirBorrowckCtxtPreDataflow<'a, 'tcx, BD>
+    where BD: BitDenotation, BD::Bit: Debug, BD::Ctxt: HasMoveData<'tcx>
+{
+    fn path(context: &str, prepost: &str, path: &str) -> PathBuf {
```

Comment:
It made sense in my head at the time (since its a helper for just the two other fn's in that same `impl` block), but its easy enough to move it out of the `impl`